src/gsy_e/gsy_e_core/simulation/simulation.py

```diff
@@ -438,10 +448,24 @@ def _execute_simulation(
             self.progress_info.update(slot_no, slot_count, self._time, self.config)
 
             self.area.cycle_coefficients_trading(self.progress_info.current_slot_time)
+            log.error(self.area.current_market_time_slot)
+
+            if self.config.external_connection_enabled:
+                global_objects.scm_external_global_stats.update()
+                self.area.publish_market_cycle_to_external_clients()
 
             global_objects.profiles_handler.update_time_and_buffer_profiles(
                 self._get_current_market_time_slot(slot_no), area=self.area)
 
+            if self.config.external_connection_enabled:
+
+                self.config.external_redis_communicator.approve_aggregator_commands()
+
+                self.area.market_cycle_external()
+
+                self.config.external_redis_communicator.\
+                    publish_aggregator_commands_responses_events()
+
```

Comment:
Hmmm, I do not like it that much that external connection handling is in the main simulation method. Can these if blocks not reside in a separate class? For instance, I think that all these can be wrapped around a class that we will only have to call once and execute all these methods in the class body instead of the _execute_simulation.

Reply:
I played around a lot with the order of these (also mixed them up with the the other commands in this method). After finding the right order i wanted to refactor this into a separate method, but forgot. I agree, lets even add a class that deals with the external connections on the SCM simulations.

Reply:
As the external communication is done a bit differently in the normal simulation I decided to only move these steps into a separate method in the SCMSimulation: [0d2cbfc](https://github.com/gridsingularity/gsy-e/pull/1559/commits/0d2cbfcb410817f89992821aed22687ae80b80af)
Please let me know what you think. Thanks